stress: failed test in cockroach/testutils/testcluster/testcluster.test: TestStopServer

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/613e0865daf0bed9404170caec18db57362112e7

Stress build found a failed test:

```
=== RUN   TestStopServer
I160901 05:26:46.387931 1709 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160901 05:26:46.391245 1709 gossip/gossip.go:230  [n?] initial resolvers: []
W160901 05:26:46.394072 1709 gossip/gossip.go:1041  [n?] no resolvers found; use --join to specify a connected node
W160901 05:26:46.405792 1709 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I160901 05:26:46.406440 1709 server/node.go:413  [n?] store store=0:0 not bootstrapped
I160901 05:26:46.408579 1841 storage/replica_trigger.go:308  new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 409085h26m55.408102132s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=2016-09-01 05:26:46.408547354 +0000 UTC]
I160901 05:26:46.409164 1709 server/node.go:345  [n?] **** cluster {c49e4249-1ee8-4ab2-ab32-d8ed3f39596a} has been created
I160901 05:26:46.409190 1709 server/node.go:346  [n?] **** add additional nodes by specifying --join=127.0.0.1:37339
I160901 05:26:46.409424 1709 gossip/gossip.go:264  [n?] NodeID set to 1
I160901 05:26:46.409898 1709 storage/store.go:980  [n?] store=1:1: failed initial metrics computation: store=1:1: system config not yet available
I160901 05:26:46.409922 1709 server/node.go:426  [n?] initialized store store=1:1: {Capacity:536870912 Available:536870912 RangeCount:0}
I160901 05:26:46.409959 1709 server/node.go:318  [n?] node ID 1 initialized
I160901 05:26:46.409996 1709 gossip/gossip.go:278  [n?] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37339" > attrs:<> 
I160901 05:26:46.410041 1709 storage/stores.go:313  read 0 node addresses from persistent storage
I160901 05:26:46.410083 1709 server/node.go:554  [n?] connecting to gossip network to verify cluster ID...
I160901 05:26:46.410213 1709 server/node.go:575  [n?] node connected via gossip and verified as part of cluster {"c49e4249-1ee8-4ab2-ab32-d8ed3f39596a"}
I160901 05:26:46.410232 1709 server/node.go:364  [n?] node=1: started with [[]=] engine(s) and attributes []
I160901 05:26:46.410251 1709 server/server.go:488  [n1] starting https server at 127.0.0.1:41730
I160901 05:26:46.410261 1709 server/server.go:489  [n1] starting grpc/postgres server at 127.0.0.1:37339
I160901 05:26:46.410325 1890 storage/split_queue.go:99  splitting store=1:1 1{/Min-/Max} at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160901 05:26:46.410476 1884 sql/event_log.go:109  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:37339} Attrs:} ClusterID:{UUID:c49e4249-1ee8-4ab2-ab32-d8ed3f39596a} StartedAt:1472707606410217759}
I160901 05:26:46.412638 1884 sql/event_log.go:109  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:37339} Attrs:} ClusterID:{UUID:c49e4249-1ee8-4ab2-ab32-d8ed3f39596a} StartedAt:1472707606410217759}
I160901 05:26:46.414050 1890 storage/replica_command.go:2213  [n1,s1,r1{/Min-/Max}] initiating a split of this range at key /Table/11
E160901 05:26:46.424775 1892 storage/queue.go:516  [replicate] (purgatory) on store=1:1 1{/Min-/Table/11}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
E160901 05:26:46.424868 1923 storage/queue.go:516  [replicate] (purgatory) on store=1:1 1{/Min-/Table/11}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
I160901 05:26:46.425443 1890 storage/replica_command.go:2213  [n1,s1,r2{/Table/11-/Max}] initiating a split of this range at key /Table/12
E160901 05:26:46.429894 1892 storage/queue.go:516  [replicate] (purgatory) on store=1:1 2{/Table/11-/Table/12}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
I160901 05:26:46.430394 1890 storage/replica_command.go:2213  [n1,s1,r3{/Table/12-/Max}] initiating a split of this range at key /Table/13
W160901 05:26:46.430633 1900 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x93"/"rdsc\x00" must be greater than start /Local/Range/"\x93"/RangeDescriptor
E160901 05:26:46.435493 1892 storage/queue.go:516  [replicate] (purgatory) on store=1:1 3{/Table/12-/Table/13}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
I160901 05:26:46.436560 1890 storage/replica_command.go:2213  [n1,s1,r4{/Table/13-/Max}] initiating a split of this range at key /Table/14
W160901 05:26:46.437141 1902 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x94"/"rdsc\x00" must be greater than start /Local/Range/"\x94"/RangeDescriptor
E160901 05:26:46.442554 1892 storage/queue.go:516  [replicate] (purgatory) on store=1:1 4{/Table/13-/Table/14}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
E160901 05:26:46.443779 1892 storage/queue.go:516  [replicate] (purgatory) on store=1:1 5{/Table/14-/Max}: 0 of 1 store with an attribute matching []; likely not enough nodes in cluster
W160901 05:26:46.443816 1904 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x95"/"rdsc\x00" must be greater than start /Local/Range/"\x95"/RangeDescriptor
I160901 05:26:46.475125 1709 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160901 05:26:46.477912 1709 gossip/gossip.go:230  [n?] initial resolvers: [%!s(*resolver.socketResolver=&{tcp 127.0.0.1:37339})]
W160901 05:26:46.477949 1709 gossip/gossip.go:1043  [n?] no incoming or outgoing connections
W160901 05:26:46.478385 1709 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I160901 05:26:46.478829 1709 server/node.go:413  [n?] store store=0:0 not bootstrapped
I160901 05:26:46.478853 1709 storage/stores.go:313  read 0 node addresses from persistent storage
I160901 05:26:46.478875 1709 server/node.go:554  [n?] connecting to gossip network to verify cluster ID...
I160901 05:26:46.499584 1963 gossip/client.go:125  [n?] node 0: started gossip client to 127.0.0.1:37339
I160901 05:26:46.500132 1942 gossip/server.go:274  [n1] node 1: received gossip from unknown node
I160901 05:26:46.500475 1709 server/node.go:575  [n?] node connected via gossip and verified as part of cluster {"c49e4249-1ee8-4ab2-ab32-d8ed3f39596a"}
I160901 05:26:46.500795 1943 storage/stores.go:328  wrote 1 node addresses to persistent storage
I160901 05:26:46.502558 1709 kv/dist_sender.go:344  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I160901 05:26:46.503419 1709 server/node.go:312  [n?] new node allocated ID 2
I160901 05:26:46.503447 1709 gossip/gossip.go:264  [n?] NodeID set to 2
I160901 05:26:46.503511 1709 gossip/gossip.go:278  [n?] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:37166" > attrs:<> 
I160901 05:26:46.503582 1709 server/node.go:364  [n?] node=2: started with [[]=] engine(s) and attributes []
I160901 05:26:46.503618 1709 server/server.go:488  [n2] starting https server at 127.0.0.1:36307
I160901 05:26:46.503638 1709 server/server.go:489  [n2] starting grpc/postgres server at 127.0.0.1:37166
I160901 05:26:46.504588 2005 sql/event_log.go:109  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:37166} Attrs:} ClusterID:{UUID:c49e4249-1ee8-4ab2-ab32-d8ed3f39596a} StartedAt:1472707606503561500}
I160901 05:26:46.538553 1990 storage/stores.go:328  wrote 1 node addresses to persistent storage
I160901 05:26:46.548577 1709 storage/engine/rocksdb.go:354  opening in memory rocksdb instance
I160901 05:26:46.549561 2002 gossip/gossip.go:264  [n2] NodeID set to 2
I160901 05:26:46.549705 2002 server/node.go:535  [n2] bootstrapped store store=2:2
I160901 05:26:46.559785 1709 gossip/gossip.go:230  [n?] initial resolvers: [%!s(*resolver.socketResolver=&{tcp 127.0.0.1:37339})]
W160901 05:26:46.559817 1709 gossip/gossip.go:1043  [n?] no incoming or outgoing connections
W160901 05:26:46.560247 1709 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I160901 05:26:46.560822 1709 server/node.go:413  [n?] store store=0:0 not bootstrapped
I160901 05:26:46.560844 1709 storage/stores.go:313  read 0 node addresses from persistent storage
I160901 05:26:46.560863 1709 server/node.go:554  [n?] connecting to gossip network to verify cluster ID...
I160901 05:26:46.582518 2102 gossip/client.go:125  [n?] node 0: started gossip client to 127.0.0.1:37339
I160901 05:26:46.583183 2020 gossip/server.go:274  [n1] node 1: received gossip from unknown node
I160901 05:26:46.587780 1709 server/node.go:575  [n?] node connected via gossip and verified as part of cluster {"c49e4249-1ee8-4ab2-ab32-d8ed3f39596a"}
I160901 05:26:46.587889 1249 storage/stores.go:328  wrote 1 node addresses to persistent storage
I160901 05:26:46.587938 1249 storage/stores.go:328  wrote 2 node addresses to persistent storage
I160901 05:26:46.589123 1980 storage/replica_raftstorage.go:492  generated snapshot for range 1 at index 36 in 1.718608ms. encoded size=118757, 417 KV pairs, 26 log entries
I160901 05:26:46.590943 2124 storage/replica_raftstorage.go:631  store=2:2 1{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 36 (encoded size=118757, 417 KV pairs, 26 log entries)
I160901 05:26:46.592639 2124 storage/replica_raftstorage.go:634  store=2:2 1{/Min-/Table/11}: with replicaID [?], applied preemptive snapshot in 0.002s
I160901 05:26:46.598946 1709 kv/dist_sender.go:344  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I160901 05:26:46.599377 1923 storage/replica_command.go:3073  change replicas of 1: read existing descriptor range_id:1 start_key:"" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160901 05:26:46.600005 1709 server/node.go:312  [n?] new node allocated ID 3
I160901 05:26:46.600024 1709 gossip/gossip.go:264  [n?] NodeID set to 3
I160901 05:26:46.600055 1709 gossip/gossip.go:278  [n?] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:40560" > attrs:<> 
I160901 05:26:46.600131 1709 server/node.go:364  [n?] node=3: started with [[]=] engine(s) and attributes []
I160901 05:26:46.600148 1709 server/server.go:488  [n3] starting https server at 127.0.0.1:33735
I160901 05:26:46.600158 1709 server/server.go:489  [n3] starting grpc/postgres server at 127.0.0.1:40560
I160901 05:26:46.601118 2138 sql/event_log.go:109  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:40560} Attrs:} ClusterID:{UUID:c49e4249-1ee8-4ab2-ab32-d8ed3f39596a} StartedAt:1472707606600116408}
I160901 05:26:46.605516 1916 storage/stores.go:328  wrote 2 node addresses to persistent storage
I160901 05:26:46.605972 1917 storage/stores.go:328  wrote 2 node addresses to persistent storage
I160901 05:26:46.620081 1923 storage/replica.go:1551  [n1,s1,r1{/Min-/Table/11}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2} for range 1: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I160901 05:26:46.620784 2135 gossip/gossip.go:264  [n3] NodeID set to 3
I160901 05:26:46.620971 2135 server/node.go:535  [n3] bootstrapped store store=3:3
W160901 05:26:46.623617 2088 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/""/"rdsc\x00" must be greater than start /Local/Range/""/RangeDescriptor
I160901 05:26:46.624143 2230 storage/replica_raftstorage.go:492  generated snapshot for range 2 at index 14 in 126.688µs. encoded size=2678, 10 KV pairs, 4 log entries
I160901 05:26:46.624804 2129 storage/replica_raftstorage.go:631  store=2:2 2{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 14 (encoded size=2678, 10 KV pairs, 4 log entries)
I160901 05:26:46.625080 2129 storage/replica_raftstorage.go:634  store=2:2 2{/Table/11-/Table/12}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.625573 1923 storage/replica_command.go:3073  change replicas of 2: read existing descriptor range_id:2 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160901 05:26:46.625656 2244 storage/raft_transport.go:378  raft transport stream to node 1 established
I160901 05:26:46.628718 2231 storage/replica_trigger.go:458  unable to gossip first range; hasLease=false, err=<nil>
I160901 05:26:46.630579 1923 storage/replica.go:1551  [n1,s1,r2{/Table/11-/Table/12}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2} for range 2: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
W160901 05:26:46.633512 2081 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x93"/"rdsc\x00" must be greater than start /Local/Range/"\x93"/RangeDescriptor
I160901 05:26:46.663490 2265 storage/replica_raftstorage.go:492  generated snapshot for range 3 at index 18 in 199.1µs. encoded size=6682, 25 KV pairs, 8 log entries
I160901 05:26:46.664065 1921 storage/replica_raftstorage.go:631  store=3:3 3{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 18 (encoded size=6682, 25 KV pairs, 8 log entries)
I160901 05:26:46.664787 1921 storage/replica_raftstorage.go:634  store=3:3 3{/Table/12-/Table/13}: with replicaID [?], applied preemptive snapshot in 0.001s
I160901 05:26:46.665251 1923 storage/replica_command.go:3073  change replicas of 3: read existing descriptor range_id:3 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160901 05:26:46.668334 1923 storage/replica.go:1551  [n1,s1,r3{/Table/12-/Table/13}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2} for range 3: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I160901 05:26:46.670166 2279 storage/replica_raftstorage.go:492  generated snapshot for range 4 at index 19 in 581.682µs. encoded size=10311, 54 KV pairs, 9 log entries
W160901 05:26:46.670413 2278 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x94"/"rdsc\x00" must be greater than start /Local/Range/"\x94"/RangeDescriptor
I160901 05:26:46.670683 2267 storage/replica_raftstorage.go:631  store=3:3 4{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 19 (encoded size=10311, 54 KV pairs, 9 log entries)
I160901 05:26:46.670961 2267 storage/replica_raftstorage.go:634  store=3:3 4{/Table/13-/Table/14}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.671376 1923 storage/replica_command.go:3073  change replicas of 4: read existing descriptor range_id:4 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160901 05:26:46.674314 1923 storage/replica.go:1551  [n1,s1,r4{/Table/13-/Table/14}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2} for range 4: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I160901 05:26:46.675705 2293 storage/replica_raftstorage.go:492  generated snapshot for range 5 at index 11 in 95.774µs. encoded size=531, 8 KV pairs, 1 log entries
W160901 05:26:46.675903 2291 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x95"/"rdsc\x00" must be greater than start /Local/Range/"\x95"/RangeDescriptor
I160901 05:26:46.676185 2294 storage/replica_raftstorage.go:631  store=2:2 5{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 11 (encoded size=531, 8 KV pairs, 1 log entries)
I160901 05:26:46.676476 2294 storage/replica_raftstorage.go:634  store=2:2 5{/Table/14-/Max}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.679714 1923 storage/replica_command.go:3073  change replicas of 5: read existing descriptor range_id:5 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160901 05:26:46.683366 2282 storage/raft_transport.go:378  raft transport stream to node 1 established
I160901 05:26:46.685365 1923 storage/replica.go:1551  [n1,s1,r5{/Table/14-/Max}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2} for range 5: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I160901 05:26:46.686313 1923 storage/queue.go:562  [replicate] purgatory is now empty
I160901 05:26:46.688706 2272 storage/replica_raftstorage.go:492  generated snapshot for range 1 at index 52 in 1.66407ms. encoded size=173247, 582 KV pairs, 42 log entries
W160901 05:26:46.696117 2285 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x96"/"rdsc\x00" must be greater than start /Local/Range/"\x96"/RangeDescriptor
I160901 05:26:46.698447 2287 storage/replica_raftstorage.go:631  store=3:3 1{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 52 (encoded size=173247, 582 KV pairs, 42 log entries)
I160901 05:26:46.699650 2287 storage/replica_raftstorage.go:634  store=3:3 1{/Min-/Table/11}: with replicaID [?], applied preemptive snapshot in 0.001s
I160901 05:26:46.700148 1892 storage/replica_command.go:3073  change replicas of 1: read existing descriptor range_id:1 start_key:"" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I160901 05:26:46.703759 1892 storage/replica.go:1551  [n1,s1,r1{/Min-/Table/11}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3} for range 1: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I160901 05:26:46.705344 2224 storage/replica_trigger.go:458  unable to gossip first range; hasLease=false, err=<nil>
W160901 05:26:46.706052 2306 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/""/"rdsc\x00" must be greater than start /Local/Range/""/RangeDescriptor
I160901 05:26:46.706106 2297 storage/replica_raftstorage.go:492  generated snapshot for range 2 at index 16 in 230.687µs. encoded size=3893, 12 KV pairs, 6 log entries
I160901 05:26:46.706696 2324 storage/replica_raftstorage.go:631  store=3:3 2{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 16 (encoded size=3893, 12 KV pairs, 6 log entries)
I160901 05:26:46.707015 2324 storage/replica_raftstorage.go:634  store=3:3 2{/Table/11-/Table/12}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.707696 1892 storage/replica_command.go:3073  change replicas of 2: read existing descriptor range_id:2 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I160901 05:26:46.717038 2298 storage/replica_trigger.go:458  unable to gossip first range; hasLease=false, err=<nil>
I160901 05:26:46.717823 1892 storage/replica.go:1551  [n1,s1,r2{/Table/11-/Table/12}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3} for range 2: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
W160901 05:26:46.720947 2299 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x93"/"rdsc\x00" must be greater than start /Local/Range/"\x93"/RangeDescriptor
I160901 05:26:46.721363 2151 storage/replica_raftstorage.go:492  generated snapshot for range 5 at index 13 in 135.89µs. encoded size=1751, 10 KV pairs, 3 log entries
I160901 05:26:46.722982 2301 storage/replica_raftstorage.go:631  store=3:3 5{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 13 (encoded size=1751, 10 KV pairs, 3 log entries)
I160901 05:26:46.723292 2301 storage/replica_raftstorage.go:634  store=3:3 5{/Table/14-/Max}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.723746 1892 storage/replica_command.go:3073  change replicas of 5: read existing descriptor range_id:5 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I160901 05:26:46.728867 1892 storage/replica.go:1551  [n1,s1,r5{/Table/14-/Max}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3} for range 5: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I160901 05:26:46.731773 2326 storage/replica_raftstorage.go:492  generated snapshot for range 4 at index 30 in 323.331µs. encoded size=19955, 81 KV pairs, 20 log entries
I160901 05:26:46.732405 2232 storage/replica_raftstorage.go:631  store=2:2 4{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 30 (encoded size=19955, 81 KV pairs, 20 log entries)
I160901 05:26:46.732906 2232 storage/replica_raftstorage.go:634  store=2:2 4{/Table/13-/Table/14}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.733620 1892 storage/replica_command.go:3073  change replicas of 4: read existing descriptor range_id:4 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
W160901 05:26:46.734536 2302 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x96"/"rdsc\x00" must be greater than start /Local/Range/"\x96"/RangeDescriptor
I160901 05:26:46.741735 1892 storage/replica.go:1551  [n1,s1,r4{/Table/13-/Table/14}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3} for range 4: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
W160901 05:26:46.743953 2255 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x95"/"rdsc\x00" must be greater than start /Local/Range/"\x95"/RangeDescriptor
I160901 05:26:46.744764 2303 storage/replica_raftstorage.go:492  generated snapshot for range 3 at index 20 in 178.166µs. encoded size=7897, 27 KV pairs, 10 log entries
I160901 05:26:46.745509 2152 storage/replica_raftstorage.go:631  store=2:2 3{/Min-/Min}: with replicaID [?], applying preemptive snapshot at index 20 (encoded size=7897, 27 KV pairs, 10 log entries)
I160901 05:26:46.745890 2152 storage/replica_raftstorage.go:634  store=2:2 3{/Table/12-/Table/13}: with replicaID [?], applied preemptive snapshot in 0.000s
I160901 05:26:46.746626 1892 storage/replica_command.go:3073  change replicas of 3: read existing descriptor range_id:3 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I160901 05:26:46.754751 1892 storage/replica.go:1551  [n1,s1,r3{/Table/12-/Table/13}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3} for range 3: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
W160901 05:26:46.767163 2225 storage/intent_resolver.go:344  could not GC completed transaction: end key /Local/Range/"\x94"/"rdsc\x00" must be greater than start /Local/Range/"\x94"/RangeDescriptor
I160901 05:26:46.963039 1976 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160901 05:26:46.963094 2153 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160901 05:26:46.963101 1974 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:37166->127.0.0.1:38565: use of closed network connection
I160901 05:26:46.963110 1239 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
W160901 05:26:46.963189 2244 storage/raft_transport.go:383  raft transport stream to node 1 failed: rpc error: code = 13 desc = transport is closing
I160901 05:26:46.963207 2012 /go/src/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:37166: operation was canceled"; Reconnecting to {"127.0.0.1:37166" <nil>}
I160901 05:26:46.963221 2012 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: grpc: the connection is closing
W160901 05:26:46.963228 2170 storage/raft_transport.go:383  raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
I160901 05:26:46.963287 2345 /go/src/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 127.0.0.1:38828->127.0.0.1:37166: write: connection reset by peer"; Reconnecting to {"127.0.0.1:37166" <nil>}
I160901 05:26:46.974456 1729 /go/src/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: write tcp 127.0.0.1:38826->127.0.0.1:37166: write: broken pipe"; Reconnecting to {"127.0.0.1:37166" <nil>}
I160901 05:26:46.990386 2345 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160901 05:26:46.990689 1898 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:37339->127.0.0.1:55468: use of closed network connection
I160901 05:26:46.990728 1924 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
W160901 05:26:46.990858 2282 storage/raft_transport.go:383  raft transport stream to node 1 failed: EOF
I160901 05:26:46.990977 1242 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160901 05:26:46.991046 2109 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:37339->127.0.0.1:55614: use of closed network connection
I160901 05:26:46.991056 1729 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160901 05:26:46.991179 2104 /go/src/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:37339: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:37339" <nil>}
I160901 05:26:46.991628 2104 /go/src/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: context canceled
I160901 05:26:46.991638 2168 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:40560->127.0.0.1:33168: use of closed network connection
--- FAIL: TestStopServer (0.61s)
	testcluster_test.go:226: Expected error from server with text "connection refused", got error with text "status: 404 Not Found, body: 404 page not found\n"
FAIL


ERROR: exit status 1
```

Run Details:

```
32 runs so far, 0 failures, over 5s
65 runs so far, 0 failures, over 10s
109 runs so far, 0 failures, over 15s
142 runs so far, 0 failures, over 20s
180 runs so far, 0 failures, over 25s
221 runs so far, 0 failures, over 30s
257 runs so far, 0 failures, over 35s
297 runs so far, 0 failures, over 40s
336 runs so far, 0 failures, over 45s
371 runs so far, 0 failures, over 50s
408 runs so far, 0 failures, over 55s
446 runs so far, 0 failures, over 1m0s
486 runs so far, 0 failures, over 1m5s
522 runs so far, 0 failures, over 1m10s
563 runs so far, 0 failures, over 1m15s
600 runs so far, 0 failures, over 1m20s
637 runs so far, 0 failures, over 1m25s
676 runs so far, 0 failures, over 1m30s
713 runs so far, 0 failures, over 1m35s
752 runs so far, 0 failures, over 1m40s
791 runs so far, 0 failures, over 1m45s
829 runs so far, 0 failures, over 1m50s
868 runs so far, 0 failures, over 1m55s
904 runs so far, 0 failures, over 2m0s
944 runs so far, 0 failures, over 2m5s
982 runs so far, 0 failures, over 2m10s
1019 runs so far, 0 failures, over 2m15s
1058 runs so far, 0 failures, over 2m20s
1094 runs so far, 0 failures, over 2m25s
1133 runs so far, 0 failures, over 2m30s
1171 runs so far, 0 failures, over 2m35s
1210 runs so far, 0 failures, over 2m40s
1246 runs so far, 0 failures, over 2m45s
1287 runs so far, 0 failures, over 2m50s
1324 runs so far, 0 failures, over 2m55s
1361 runs so far, 0 failures, over 3m0s
1401 runs so far, 0 failures, over 3m5s
1435 runs so far, 0 failures, over 3m10s
1476 runs so far, 0 failures, over 3m15s
1514 runs so far, 0 failures, over 3m20s
1551 runs so far, 0 failures, over 3m25s
1590 runs so far, 0 failures, over 3m30s
1625 runs so far, 0 failures, over 3m35s
1668 runs so far, 0 failures, over 3m40s
1703 runs so far, 0 failures, over 3m45s
1740 runs so far, 0 failures, over 3m50s
1782 runs so far, 0 failures, over 3m55s
1816 runs so far, 0 failures, over 4m0s
1857 runs so far, 0 failures, over 4m5s
1894 runs so far, 0 failures, over 4m10s
1934 runs so far, 0 failures, over 4m15s
1972 runs so far, 0 failures, over 4m20s
2008 runs so far, 0 failures, over 4m25s
2047 runs so far, 0 failures, over 4m30s
2085 runs so far, 0 failures, over 4m35s
2121 runs so far, 0 failures, over 4m40s
2161 runs so far, 0 failures, over 4m45s
2198 runs so far, 0 failures, over 4m50s
2239 runs so far, 0 failures, over 4m55s
2273 runs so far, 0 failures, over 5m0s
2315 runs so far, 0 failures, over 5m5s
2352 runs so far, 0 failures, over 5m10s
2389 runs so far, 0 failures, over 5m15s
2427 runs so far, 0 failures, over 5m20s
2466 runs so far, 0 failures, over 5m25s
2503 runs so far, 0 failures, over 5m30s
2542 runs so far, 0 failures, over 5m35s
2580 runs so far, 0 failures, over 5m40s
2620 runs so far, 0 failures, over 5m45s
2656 runs so far, 0 failures, over 5m50s
2695 runs so far, 0 failures, over 5m55s
2734 runs so far, 0 failures, over 6m0s
2770 runs so far, 0 failures, over 6m5s
2810 runs so far, 0 failures, over 6m10s
2847 runs so far, 0 failures, over 6m15s
2885 runs so far, 0 failures, over 6m20s
2921 runs so far, 0 failures, over 6m25s
2959 runs so far, 0 failures, over 6m30s
3000 runs so far, 0 failures, over 6m35s
3036 runs so far, 0 failures, over 6m40s
3077 runs so far, 0 failures, over 6m45s
3113 runs so far, 0 failures, over 6m50s
3150 runs so far, 0 failures, over 6m55s
3191 runs so far, 0 failures, over 7m0s
3229 runs so far, 0 failures, over 7m5s
3265 runs so far, 0 failures, over 7m10s
3304 runs so far, 0 failures, over 7m15s
3340 runs so far, 0 failures, over 7m20s
3381 runs so far, 0 failures, over 7m25s
3418 runs so far, 0 failures, over 7m30s
3457 runs so far, 0 failures, over 7m35s
3494 runs so far, 0 failures, over 7m40s
3533 runs so far, 0 failures, over 7m45s
3570 runs so far, 0 failures, over 7m50s
3607 runs so far, 0 failures, over 7m55s
3647 runs so far, 0 failures, over 8m0s
3684 runs so far, 0 failures, over 8m5s
3724 runs so far, 0 failures, over 8m10s
3757 runs so far, 0 failures, over 8m15s
3798 runs so far, 0 failures, over 8m20s
3837 runs so far, 0 failures, over 8m25s
3874 runs so far, 0 failures, over 8m30s
3913 runs so far, 0 failures, over 8m35s
3950 runs so far, 0 failures, over 8m40s
3991 runs so far, 0 failures, over 8m45s
4024 runs so far, 0 failures, over 8m50s
4063 runs so far, 0 failures, over 8m55s
4101 runs so far, 0 failures, over 9m0s
4107 runs completed, 1 failures, over 9m1s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/gossip/client_test.go
+++ b/gossip/client_test.go
@@ -41,7 +41,7 @@ import (
 
 // startGossip creates and starts a gossip instance.
 func startGossip(nodeID roachpb.NodeID, stopper *stop.Stopper, t *testing.T, registry *metric.Registry) *Gossip {
-	return startGossipAtAddr(nodeID, util.TestAddr, stopper, t, registry)
+	return startGossipAtAddr(nodeID, util.IsolatedTestAddr, stopper, t, registry)
 }
 
 func startGossipAtAddr(nodeID roachpb.NodeID, addr net.Addr, stopper *stop.Stopper, t *testing.T, registry *metric.Registry) *Gossip {
@@ -109,7 +109,7 @@ func startFakeServerGossips(t *testing.T) (*Gossip, *fakeGossipServer, *stop.Sto
 
 	lserver := rpc.NewServer(lRPCContext)
 	local := New(context.TODO(), lRPCContext, lserver, nil, stopper, metric.NewRegistry())
-	lln, err := netutil.ListenAndServeGRPC(stopper, lserver, util.TestAddr)
+	lln, err := netutil.ListenAndServeGRPC(stopper, lserver, util.IsolatedTestAddr)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -118,7 +118,7 @@ func startFakeServerGossips(t *testing.T) (*Gossip, *fakeGossipServer, *stop.Sto
 	rRPCContext := rpc.NewContext(context.TODO(), &base.Context{Insecure: true}, nil, stopper)
 
 	rserver := rpc.NewServer(rRPCContext)
-	rln, err := netutil.ListenAndServeGRPC(stopper, rserver, util.TestAddr)
+	rln, err := netutil.ListenAndServeGRPC(stopper, rserver, util.IsolatedTestAddr)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -399,7 +399,7 @@ func TestClientRegisterWithInitNodeID(t *testing.T) {
 		RPCContext := rpc.NewContext(context.TODO(), &base.Context{Insecure: true}, nil, stopper)
 
 		server := rpc.NewServer(RPCContext)
-		ln, err := netutil.ListenAndServeGRPC(stopper, server, util.TestAddr)
+		ln, err := netutil.ListenAndServeGRPC(stopper, server, util.IsolatedTestAddr)
 		if err != nil {
 			t.Fatal(err)
 		}

--- a/server/testserver.go
+++ b/server/testserver.go
@@ -78,10 +78,11 @@ func makeTestContext() Context {
 	ctx.SSLCertKey = filepath.Join(security.EmbeddedCertsDir, security.EmbeddedNodeKey)
 
 	// Addr defaults to localhost with port set at time of call to
-	// Start() to an available port.
-	// Call TestServer.ServingAddr() for the full address (including bound port).
-	ctx.Addr = "127.0.0.1:0"
-	ctx.HTTPAddr = "127.0.0.1:0"
+	// Start() to an available port. May be overridden later (as in
+	// makeTestContextFromParams). Call TestServer.ServingAddr() for the
+	// full address (including bound port).
+	ctx.Addr = util.TestAddr.String()
+	ctx.HTTPAddr = util.TestAddr.String()
 	// Set standard user for intra-cluster traffic.
 	ctx.User = security.NodeUser
 
@@ -122,6 +123,16 @@ func makeTestContextFromParams(params base.TestServerArgs) Context {
 		ctx.EventLogEnabled = false
 	}
 	ctx.JoinList = []string{params.JoinAddr}
+	if ctx.Insecure {
+		// Whenever we can (i.e. in insecure mode), use IsolatedTestAddr
+		// to prevent issues that can occur when running a test under
+		// stress.
+		ctx.Addr = util.IsolatedTestAddr.String()
+		ctx.HTTPAddr = util.IsolatedTestAddr.String()
+	} else {
+		ctx.Addr = util.TestAddr.String()
+		ctx.HTTPAddr = util.TestAddr.String()
+	}
 	return ctx
 }
 

--- a/testutils/testcluster/testcluster_test.go
+++ b/testutils/testcluster/testcluster_test.go
@@ -182,7 +182,15 @@ func TestWaitForFullReplication(t *testing.T) {
 func TestStopServer(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	tc := StartTestCluster(t, 3, base.TestClusterArgs{ReplicationMode: base.ReplicationAuto})
+	// Use insecure mode so our servers listen on util.IsolatedTestAddr
+	// and they fail cleanly instead of interfering with other tests.
+	// See https://github.com/cockroachdb/cockroach/issues/9256
+	tc := StartTestCluster(t, 3, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			Insecure: true,
+		},
+		ReplicationMode: base.ReplicationAuto,
+	})
 	defer tc.Stopper().Stop()
 	if err := tc.WaitForFullReplication(); err != nil {
 		t.Fatal(err)

--- a/util/testaddr_default.go
+++ b/util/testaddr_default.go
@@ -1,0 +1,24 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Ben Darnell
+
+// +build !linux
+
+package util
+
+func init() {
+	IsolatedTestAddr = TestAddr
+}

--- a/util/testaddr_random.go
+++ b/util/testaddr_random.go
@@ -1,0 +1,36 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Ben Darnell
+
+// +build linux
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/util/randutil"
+)
+
+func init() {
+	r, _ := randutil.NewPseudoRand()
+	// 127.255.255.255 is special (broadcast), so choose values less
+	// than 255.
+	a := r.Intn(255)
+	b := r.Intn(255)
+	c := r.Intn(255)
+	IsolatedTestAddr = NewUnresolvedAddr("tcp", fmt.Sprintf("127.%d.%d.%d:0", a, b, c))
+}

--- a/util/unresolved_addr.go
+++ b/util/unresolved_addr.go
@@ -21,9 +21,27 @@ import (
 	"net"
 )
 
-// TestAddr is an address to use for test servers. Listening on port 0 causes
-// the kernel to allocate an unused port.
+// TestAddr is an address to use for test servers. Listening on port 0
+// causes the kernel to allocate an unused port.
 var TestAddr = NewUnresolvedAddr("tcp", "127.0.0.1:0")
+
+// IsolatedTestAddr is initialized in testaddr_*.go
+
+// IsolatedTestAddr is an address to use for tests that need extra
+// isolation by using more addresses than 127.0.0.1 (support for this
+// is platform-specific and only enabled on Linux). Both TestAddr and
+// IsolatedTestAddr guarantee that the chosen port is not in use when
+// allocated, but IsolatedTestAddr draws from a larger pool of
+// addresses so that when tests are run in a tight loop the system is
+// less likely to run out of available ports or give a port to one
+// test immediately after it was closed by another.
+//
+// IsolatedTestAddr should be used for tests that open and close a
+// large number of sockets, or tests which stop a server and rely on
+// seeing a "connection refused" error afterwards. It cannot be used
+// with tests that operate in secure mode since our test certificates
+// are only valid for 127.0.0.1.
+var IsolatedTestAddr *UnresolvedAddr
 
 // MakeUnresolvedAddr populates an UnresolvedAddr from a network and raw
 // address string.

